include/boost/geometry/core/config.hpp

```diff
@@ -13,17 +13,27 @@
 #define BOOST_GEOMETRY_CORE_CONFIG_HPP
 
 #include <boost/config.hpp>
+#include <boost/config/pragma_message.hpp>
 
-// Flip around the default. This is targeted to tests. As soon as the default changes,
-// some unit tests (tagged by _alternative) are tested with the non-default.
-#if defined(BOOST_GEOMETRY_ROBUSTNESS_ALTERNATIVE)
-  #if defined(BOOST_GEOMETRY_NO_ROBUSTNESS)
-    #undef BOOST_GEOMETRY_NO_ROBUSTNESS
+// If no define is specified, it uses BOOST_GEOMETRY_NO_ROBUSTNESS, which means: no rescaling.
+// (Rescaling was an earlier approach to fix robustness issues.)
+// However, if BOOST_GEOMETRY_ROBUSTNESS_ALTERNATIVE is specified, it flips the default.
+#if !defined(BOOST_GEOMETRY_NO_ROBUSTNESS) && !defined(BOOST_GEOMETRY_USE_RESCALING)
+  #if defined(BOOST_GEOMETRY_ROBUSTNESS_ALTERNATIVE)
+    #define BOOST_GEOMETRY_USE_RESCALING
   #else
     #define BOOST_GEOMETRY_NO_ROBUSTNESS
   #endif
 #endif
 
+#if defined(BOOST_GEOMETRY_USE_RESCALING) && ! defined(BOOST_GEOMETRY_ROBUSTNESS_ALTERNATIVE)
+    BOOST_PRAGMA_MESSAGE("Rescaling is deprecated and its functionality will be removed in Boost 1.82")
+#endif
+
+#if defined(BOOST_GEOMETRY_USE_RESCALING) && defined(BOOST_GEOMETRY_NO_ROBUSTNESS)
+    #error "Define either BOOST_GEOMETRY_NO_ROBUSTNESS (default) or BOOST_GEOMETRY_USE_RESCALING"
```

Comment:
We didn't have it before. It's similar to, for example:
https://github.com/boostorg/chrono/blob/develop/include/boost/chrono/config.hpp#L22